source/contracts/reporting/Market.sol

```diff
@@ -171,20 +169,44 @@ contract Market is DelegationTarget, Typed, Initializable, Ownable, IMarket {
     }
 
     function updateTentativeWinningPayoutDistributionHash(bytes32 _payoutDistributionHash) public returns (bool) {
+        if (_payoutDistributionHash == tentativeWinningPayoutDistributionHash) {
+            return true;
+        }
+
         IReportingToken _reportingToken = reportingTokens[_payoutDistributionHash];
         require(address(_reportingToken) != NULL_ADDRESS);
 
         IReportingToken _tentativeWinningReportingToken = reportingTokens[tentativeWinningPayoutDistributionHash];
         if (address(_tentativeWinningReportingToken) == NULL_ADDRESS) {
             tentativeWinningPayoutDistributionHash = _payoutDistributionHash;
-            _tentativeWinningReportingToken = _reportingToken;
+            return true;
         }
-        if (_reportingToken.totalSupply() > _tentativeWinningReportingToken.totalSupply()) {
+
+        int256 _tentativeWinningStake = getPayoutDistributionHashStake(tentativeWinningPayoutDistributionHash, _tentativeWinningReportingToken);
+        int256 _payoutStake = getPayoutDistributionHashStake(_payoutDistributionHash, _reportingToken);
+
+        if (_payoutStake > _tentativeWinningStake) {
+            secondPlaceTentativeWinningPayoutDistributionHash = tentativeWinningPayoutDistributionHash;
```

Comment:
I don't believe this logic is correct.  It is possible that a dispute bond placement may drop a winning outcome to 3rd place or worse, so we can't assume that just because `_payoutStake > _firstPlaceStake` that `_firstPlaceStake > _secondPlaceStake`.

Reply:
The only case where second place matters is during promotion during a `limited` dispute. When a dispute occurs if second place is promoted it is correct that it will be promoted to second place even if it is not technically, but this won't actually cause any errors. second place may simply be a bad name for the variable. It is simply the current best guess second place but should be guaranteed accurate in the one case we need to promote it.

Reply:
I wonder if we should rename the variable to `maybeSecondPlace` or something, to make it clear to the reader that they need to understand there are some caveats to the variable?

Reply:
`bestGuessSecondPlace...` ?

Reply:
`bestGuessSecondPlace` works for me.  Would be nice to have a comment above the variable declaration that gives a brief description of why it isn't always correct.